src/lib.rs

```diff
@@ -151,6 +151,22 @@ struct HelperInner {
     consumer_done: bool,
 }
 
+/// Error type for `from_env` function.
+#[derive(Debug)]
+pub enum ErrFromEnv {
+    /// There isn't env var, that describes jobserver to inherit.
+    IsNotConfigured,
+    /// Cannot connect following this process's environment.
+    PlatformSpecific {
+        /// Error.
+        err: imp::ErrFromEnv,
```

Comment:
Nit: Having os specific exposed doesn't sound great
I wonder if it is a better idea to use `io::Error` here since the return of error here is to improve logging/err msg on failure.

Reply:
Some errors aren't `io::Error`, wouldn't such error type mislead? I've chose from `imp::ErrFromEnv` and `{ description: &'static str, additional_info: String }`.

The reason of current choice. Os specific errors can be either printed or passed to `match` to be processed. Default `Debug` can be changed. If user want to process `PlatformSpecific`, probably it's better to explicitly state by using `cfg`, that code will deal with platform specific behavior.

If it would be enough in most cases to separate `IsNotConfigured` from other errors, I'd be sure in current variant, but unix `InvalidDescriptor`, according to gnu make, should be considered as a warning, not an error. So if os specific shouldn't be in public interface - I'll change this.

Reply:
> Some errors aren't io::Error, wouldn't such error type mislead? I've chose from imp::ErrFromEnv and { description: &'static str, additional_info: String }.

`io::ErrorKind::Others` can be used for these kind of errors and other crates I've seen also use `io::Error` for internal errors, so not exactly misleading.

For example, `tokio::task::JoinError` implements `impl From<JoinError> for std::io::Error`.

[`std::thread::Builder::spawn`](https://doc.rust-lang.org/std/thread/struct.Builder.html#method.spawn) also uses `io::Error` even if creating a new thread isn't an I/O.

> The reason of current choice. Os specific errors can be either printed or passed to match to be processed. Default Debug can be changed. If user want to process PlatformSpecific, probably it's better to explicitly state by using cfg, that code will deal with platform specific behavior.

Does the user really need that much flexibility?
I think if you want to create a new error type for `jobserver`, you'd better create a new `struct Error` that does not expose any public fields for the sake of backwards compatibility and instead only expose methods to return `ErrorKind` and implement `std::error::Error::source` for user to access the underlying source if possible.

> If it would be enough in most cases to separate IsNotConfigured from other errors, I'd be sure in current variant, but unix InvalidDescriptor, according to gnu make, should be considered as a warning, not an error. So if os specific shouldn't be in public interface - I'll change this.

IMO `io::ErrorKind` already provides enough kinds to classify the requirements in #51 

> Error or warn if the jobserver descriptors are open/accessible, but corrupted, i.e. reopened by someone else and no longer refer to pipes 

IMO [`io::ErrorKind::InvalidData`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.InvalidInput) can be used for this.

Reply:
I didn't know enough about `io::Error`. Thanks, I'll change to this.